helix-lsp/src/lib.rs

```diff
@@ -66,39 +66,26 @@ pub mod util {
         pos: lsp::Position,
         offset_encoding: OffsetEncoding,
     ) -> Option<usize> {
-        let max_line = doc.lines().count().saturating_sub(1);
         let pos_line = pos.line as usize;
-        let pos_line = if pos_line > max_line {
+        if pos_line > doc.len_lines() - 1 {
```

Comment:
Should still use `saturating_sub` if you have `0` lines (no line endings)

Reply:
I noticed that it would return `1` even on an empty string `""`.

Reply:
Looks good then 👍🏻